src/index.tsx

```diff
@@ -239,240 +243,241 @@ const SUPPORTS_PATH2D = (function () {
   return true;
 })();
 
-const QRCodeCanvas = React.forwardRef(function QRCodeCanvas(
-  props: QRPropsCanvas,
-  forwardedRef: React.ForwardedRef<HTMLCanvasElement>
-) {
-  const {
-    value,
-    size = DEFAULT_SIZE,
-    level = DEFAULT_LEVEL,
-    bgColor = DEFAULT_BGCOLOR,
-    fgColor = DEFAULT_FGCOLOR,
-    includeMargin = DEFAULT_INCLUDEMARGIN,
-    minVersion = DEFAULT_MINVERSION,
-    marginSize,
-    style,
-    imageSettings,
-    ...otherProps
-  } = props;
-  const imgSrc = imageSettings?.src;
-  const _canvas = useRef<HTMLCanvasElement | null>(null);
-  const _image = useRef<HTMLImageElement>(null);
-
-  // Set the local ref (_canvas) and also the forwarded ref from outside
-  const setCanvasRef = useCallback(
-    (node: HTMLCanvasElement | null) => {
-      _canvas.current = node;
-      if (typeof forwardedRef === 'function') {
-        forwardedRef(node);
-      } else if (forwardedRef) {
-        forwardedRef.current = node;
-      }
-    },
-    [forwardedRef]
-  );
-
-  // We're just using this state to trigger rerenders when images load. We
-  // Don't actually read the value anywhere. A smarter use of useEffect would
-  // depend on this value.
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const [isImgLoaded, setIsImageLoaded] = useState(false);
-
-  const {margin, cells, numCells, calculatedImageSettings} = useQRCode({
-    value,
-    level,
-    minVersion,
-    includeMargin,
-    marginSize,
-    imageSettings,
-    size,
-  });
+const QRCodeCanvas = React.forwardRef<HTMLCanvasElement, QRPropsCanvas>(
+  function QRCodeCanvas(props, forwardedRef) {
```

Comment:
I set the `props` and `forwardedRef` types of `QRCodeCanvas` to  `forwardRef generic`. 
QRCodeSVG is the same.

Reply:
You'll see a lot of changes tracked, but nothing really big.